RELEASE/scripts/autoscend/auto_restore.ash

```diff
@@ -351,466 +351,596 @@ string[int] __RANKED_GOAL_DESCRIPTIONS = {
+		float per_use = get_value(resource_type, "restored_per_use");
 
-  float total_uses_needed(){
-    return max(get_value("hp", "uses_needed_for_goal"), get_value("mp", "uses_needed_for_goal"));
-  }
+		if(per_use < 1.0)
+		{
+			return 0.0;
+		}
+		return max(ceil((goal - starting) / per_use), 1.0);
+	}
 
-  float meat_per_use() {
-    if (metadata.type == "item") {
-      item i = to_item(metadata.name);
-      int price = npc_price(i);
-      if (can_interact()){
-        price = min(price, auto_mall_price(i));
-      }
-      return price;
-    } else if (metadata.type == "skill") {
-      float meat_per_mp = 9.0; // default to Doc Galaktik's Invigorating Tonic at 90 meat/10 MP
-      if (dispensary_available() || black_market_available()) {
-        meat_per_mp = 8.0; // Knob Goblin seltzer or Black cherry soda at 80 meat/10 MP
-      }
-      if (get_property("questM24Doc") == "finished") {
-        meat_per_mp = 6.0; // Doc Galaktik's Invigorating Tonic reduced to 60 meat/10 MP
-      }
-      if (auto_have_skill($skill[Five Finger Discount])) {
-        meat_per_mp = meat_per_mp * 0.95; // this isn't quite right for discounted Doc Galaktik but I don't care.
-      }
-      if (isMystGuildStoreAvailable()) {
-        int mmj_cost = auto_have_skill($skill[Five Finger Discount]) ? 95 : 100;
-        int mmj_mp_restored = my_level() * 1.5 + 5;
-        float mmj_meat_per_mp = mmj_cost / mmj_mp_restored;
-        meat_per_mp = min(meat_per_mp, mmj_meat_per_mp);
-        // at level 6 and above, MMJ is better than all but discounted doc galaktik
-        // and at level 8 and above it's better than everything
-      }
-      if (my_class() == $class[Sauceror]) {
-        // your MP cup runneth over
-        meat_per_mp = 0.1;
-      }
-      skill s = to_skill(metadata.name);
-      return (mp_cost(s) * meat_per_mp);
-    } else {
-      return 0.0;
-    }
-  }
+	float total_uses_needed()
+	{
+		return max(get_value("hp", "uses_needed_for_goal"), get_value("mp", "uses_needed_for_goal"));
+	}
 
-  float tokens_per_use(){
-    if(metadata.type != "item"){
-      return 0.0;
-    }
-    item i = to_item(metadata.name);
-    if(i.seller != $coinmaster[none]){
-      return sell_price(i.seller, i);
-    }
-    return 0.0;
-  }
+	float meat_per_use()
+	{
+		if (metadata.type == "item")
+		{
+			item i = to_item(metadata.name);
+			if(can_interact())			//we have unlimited mall access = casual, aftercore, or postronin
+			{
+				int price = 999999;		//do not use items that cannot be bought
+				if(is_tradeable(i))		//is possible to trade in the mall
+				{
+					price = min(price, auto_mall_price(i));
+				}
+				if(npc_price(i) > 0)	//is possible to buy from an NPC store
+				{
+					price = min(price, npc_price(i));
+				}
+				return price;
+			}
+			else		//mall access is limited. this means pulls are limited too.
+			{
+				return npc_price(i);	//note that this sets items that cannot be purchased to free
+			}
```

Comment:
It looks to me that this is the only thing that's actually changed in this PR. Could you confirm please?